### ISSUE TITLE
JVM crashes with ILL_ILLOPC when loading libjava-unix.so on ARMv6

### ISSUE THREAD

**Issue by RafalSumislawski** [hypfvieh/libmatthew#2]
The native library for ARM is compiled for ARMv7 and contains opcodes illegal on ARMv6. ARMv6 is still fairly popular (for example Raspberry PI Zero) so it would be nice to support it. A library compiled for ARMv6 should still work on ARMv7 and any higher ARM architecture. 

I'm not the first person to hit this issue. @SStorm recompiled libjava-unix.so on ARMv6 (though his diagnosis of "segfaulting" was rather incorrect) and I managed to to run dbus-java on ARMv6 using [his version of the lib](https://github.com/imont/libmatthew/commit/451084427e71d36adcaaab625547f9ea3bcde131). 

IMO incorporating his work, without him submitting it willingly, is doubtful in terms of copyright, so I suggest that I or you (@hypfvieh) could recompile it for ARMv6. I gave first tries compiling it, but using the Makefile turned out not to be enough. Do you maybe have some hints on how to compile it?

**Comment by hypfvieh** [comment]
It has been some time since I compiled it on my raspberry pi 3. As far as I remember, make or just gcc just worked for me.

I added the ARM binary for easier usage of the lib, if this leads to more issues than it solves, we should remove the binary version.

The better approach would be to either let the user compile the lib or to notify the user to install the lib using the package manager (on debian based systems (like raspberrian) the lib is available in the apt repositories).

If you can compile it for ARMv6 I could also add this version instead of my ARMv7 as well. 
I'm fine with all solutions which helps improving the usage of this lib.

**Comment by SStorm** [comment]
Happy to submit the "patch", although there's not much there - all I did was recompile it on the Pi0 and repackage.

**Comment by RafalSumislawski** [comment]
IMO the embedded native libraries make the library convenient to use, even if they cause trouble from time to time. I've seen some popular libraries (jna, netty, sqlite-jdbc) take the same approach as you did here.  A actually migrated to `dbus-java` from `tinyb` - a library which required me to compile the native lib and it did cause a lot of trouble (recompiling also some packages it depended on).

**Comment by RafalSumislawski** [comment]
Hi @SStorm
I just submitted a PR (https://github.com/hypfvieh/libmatthew/pull/3) with a recompiled lib a few minutes ago. Funny fact: on my PI0 it wasn't problem-free.

**Comment by hypfvieh** [comment]
I know `tinyb`. I used it before and was not happy with the native compilation stuff. That's the reason why I decided to fix the original `dbus-java` to work with more modern java versions.

As the `libmatthew `is also not my own code, I did not take any further investigations to fix/cleanup the Makefile and C code part (I'm not so familiar with this stuff). The only parts I did was to fix a lot of issues related to the old code in terms of java. I also removed the 'wannabe' CGI package as it is not useful for anything today (at least in my opinion). 

I also investigated a way on removing `libmatthew `and stay with java without native code.
Sadly it is not possible to use dbus (or better unix-sockets) without a native library because java does not provide any sort of RAW sockets or other options closer to the running OS.
Switching to another unix-socket implementing native library also does not provide any benefit in my eyes.

At the end I wanted a working `dbus-java` with all dependencies managable by maven and hosted on maven central. 

So for now, I've merged the pull request of @RafalSumislawski, I hope this will help others as well.

I'll release the new versions of `libmatthew` and `dbus-java` the next few days (maybe next week).
